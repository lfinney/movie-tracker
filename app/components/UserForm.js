import React, { Component } from 'react';

export default class UserForm extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      email: '',
      password: '',
      signUp: false
    };
  }

  handleInputs(key, event) {
    this.setState({[key]: event.target.value})
  }

  render() {
    return (
      <form>
        {
          !this.props.userLoginError &&
          <div>false!</div>
        }
        {!this.state.signUp ?
          // if !signup & loginError
          // display error message
          <div>
            <input
              type="text"
              placeholder="email"
              value={this.state.email}
              onChange={ (event) => this.handleInputs('email', event) }
            />
            <input
              type="text"
              placeholder="password"
              value={this.state.password}
              onChange={ (event) => this.handleInputs('password', event) }
            />
            <button type="submit" onClick={ (event) => {
              event.preventDefault();
              this.props.verifyUserLogin(Object.assign({}, {email: this.state.email, password:this.state.password}));
            } }>Log In</button>

            <button type="submit" onClick={ (event) => {
              event.preventDefault();
              this.setState({
                signUp: true
              })
            } }>Sign Up</button>
          </div>
          :
          <div>
            <input
              type="text"
              placeholder="name"
              value={this.state.name}
              onChange={ (event) => this.handleInputs('name', event) }
            />
            <input
              type="text"
              placeholder="email"
              value={this.state.email}
              onChange={ (event) => this.handleInputs('email', event) }
            />
            <input
              type="text"
              placeholder="password"
              value={this.state.password}
              onChange={ (event) => this.handleInputs('password', event) }
            />
            <button type="submit" onClick={ (event) => {
              event.preventDefault();
              this.props.postUserSignUp(Object.assign({}, {name: this.state.name, email: this.state.email, password:this.state.password}));
            } }>Sign-Up</button>
          </div>
        }
      </form>
    )
  }
}
