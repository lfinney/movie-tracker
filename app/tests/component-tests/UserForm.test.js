import React from 'react';
import { shallow } from 'enzyme';
import UserForm from '../../components/UserForm';

describe('UserForm', () => {

  it('should render component', () => {
    const mockFunc = jest.fn();
    const wrapper = shallow(<UserForm/>);

    expect(wrapper).toMatchSnapshot();
  });
});
