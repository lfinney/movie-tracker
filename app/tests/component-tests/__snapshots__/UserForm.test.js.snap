// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm should render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <input
          onChange={[Function]}
          placeholder="email"
          type="text"
          value={undefined}
        />
        <input
          onChange={[Function]}
          placeholder="password"
          type="text"
          value={undefined}
        />
        <button
          onClick={[Function]}
          type="submit"
        >
          Log In
        </button>
        <h2
          onClick={[Function]}
        >
          Sign Up
        </h2>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <input
            onChange={[Function]}
            placeholder="email"
            type="text"
            value={undefined}
          />,
          <input
            onChange={[Function]}
            placeholder="password"
            type="text"
            value={undefined}
          />,
          <button
            onClick={[Function]}
            type="submit"
          >
            Log In
          </button>,
          <h2
            onClick={[Function]}
          >
            Sign Up
          </h2>,
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "email",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "password",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Log In",
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "Log In",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "h2",
        },
      ],
      "type": "div",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <input
            onChange={[Function]}
            placeholder="email"
            type="text"
            value={undefined}
          />
          <input
            onChange={[Function]}
            placeholder="password"
            type="text"
            value={undefined}
          />
          <button
            onClick={[Function]}
            type="submit"
          >
            Log In
          </button>
          <h2
            onClick={[Function]}
          >
            Sign Up
          </h2>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <input
              onChange={[Function]}
              placeholder="email"
              type="text"
              value={undefined}
            />,
            <input
              onChange={[Function]}
              placeholder="password"
              type="text"
              value={undefined}
            />,
            <button
              onClick={[Function]}
              type="submit"
            >
              Log In
            </button>,
            <h2
              onClick={[Function]}
            >
              Sign Up
            </h2>,
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "email",
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "password",
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Log In",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Log In",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
